Add missing i18n message documentation

samwilson opened the pull request:
Also order messages alphabetically.

Nikerabbit commented:
FYI I consider alphabetical sorting as an anti-pattern. Related messages should be close to each other.

samwilson commented:
Oh cool! Good to know. I thought it was some new trend. :) I'll switch it back though.

Is this documented somewhere?

MaxSem commented on i18n/en.json:
```diff
@@ -1,28 +1,28 @@
   "login": "Log in",
   "logout": "Log out",
-  "language-settings": "Language settings",
+  "made-by": "Made by the Wikimedia Foundation's $1",
```
I understand it wasn't introduced in this patch, however this is literally not translatable to Russian among a bunch of other languages because of cases. Patchwork sentences just never work.

samwilson replied:
I'll change it (in a separate patch). I suspect something similar applies to the three tutorial messages too (`pick-an-image` etc.) where the first part of the phrase is in a larger font and bold. Do they need changing too?

MusikAnimal replied:
"Lego" messages are bad, but mainly only if they are re-used, no? It doesn't matter if the "Community Tech" message is uniquely for `made-by`, as would be described in qqq? The idea being to avoid HTML markup https://www.mediawiki.org/wiki/Localisation#Avoid_untranslated_HTML_markup_in_messages

It's different for MediaWiki where we use wikitext. Here we have to render the message as raw, so they can put in malicious or accidentally broken markup that breaks the display.

samwilson commented:
I've fixed up all the multi-part messages, and moved their HTML into the messages themselves. See what you think.

Also updated the Toolforge logo to one without text, and added the text next to it so it can be translated.

MusikAnimal commented on i18n/en.json:
```diff
@@ -3,26 +3,29 @@
+
+  "pick-an-image": "<strong>Pick an image</strong> <span>from Commons.</span>",
+  "translate-labels": "<strong>Translate</strong> <span>the labels in the image.</span>",
+  "update-commons": "<strong>Update Commons</strong> <span>or download directly to your computer.</span>",
```
Are the spans necessary? Just leaves more room for error, say if they misspelled it or forgot the closing tag

mooeypoo replied:
This is actually bigger of an issue (I think you're talking about this elsewhere in the review, @MusikAnimal ) -- this will be "raw", and that's a pretty big security risk especially since the i18n bot self-merges into the code.

We should avoid 'raw' anywhere we can. I flagged this specific one for @Niharika29 to see if the design in this case can be flexible to not have the partial-bold part, but we should keep in mind to try and avoid html messages we display as raw and render into the DOM as much as possibly possible.

Niharika29 replied:
I think for these messages it's better to split the strings up into two parts such as `tutorial-image-head` and `tutorial-image-desc', `tutorial-translate-head` and `tutorial-translate-desc` etc. For the ones below, `licensed-under` and `made-by`, I agree with Leon that we can make them complete links and also we can switch the wording to what we use in XTools. 
Let's also document these standard wordings we use in our tools somewhere on MW so we don't have to spend time thinking about them every time and we can save @prtksxna the bother. :)

samwilson replied:
I had these in two parts in the previous iteration of this patch. I thought it was bad to split single phrases up for translation, so combined them. I guess two messages is fine if the qqq docs are clear about how the messages are going to be displayed?

The spans are necessary because the two parts of these phrases are displayed differently, and both as blocks. But I'll switch back to multiple messages, so that's moot.

samwilson replied:
Yeah, I agree with everything you say there. Done.

MusikAnimal commented on i18n/en.json:
```diff
@@ -3,26 +3,29 @@
+  "translate-icon-alt": "An icon indicating multiple languages",
+  "download-icon-alt": "An icon indicating file-download",
+
+  "licenced-under": "Licenced under <a href='https://www.gnu.org/licenses/gpl-3.0.html'>GPL 3.0 or later</a>",
```
If we word it like "Licensed under GPL 3.0+" (without _or later_) then I think we can safely inject the link as a param, "Licensed under $1", and avoid the HTML

MusikAnimal commented:
Sorry I meant to say this sooner... but did we want to import existing translations from Event Metrics, XTools, etc.? That'd give us a lot of translations for free.

You'll have to be OK with rewording a few things, but the differences are trivial:
* `made-by` (_Made by my Wikimedia Foundation's Community Tech team_) => [_Developed by Community Tech_](https://github.com/wikimedia/grantmetrics/blob/master/i18n/en.json#L32) (then the whole thing can be a link, no need to put HTML in the message)
* `report-a-bug` => [_Report an issue_](https://github.com/wikimedia/grantmetrics/blob/master/i18n/en.json#L118) (this is better anyway because not all issues are bugs)
* `view-on-github` => [_View source_](https://github.com/wikimedia/grantmetrics/blob/master/i18n/en.json#L139) (and maybe it won't always be on GitHub)

Other messages are identical from other apps:
* `powered-by-toolforge` from [InteractionTimeline](https://github.com/wikimedia/InteractionTimeline/blob/master/i18n/en.json#L28)
* `login`/`logout` from [XTools](https://github.com/x-tools/xtools/blob/master/i18n/en.json#L194-L195)
* `help` from [Pageviews Analysis](https://github.com/MusikAnimal/pageviews/blob/master/messages/en.json#L147)

You can import easily using https://github.com/wikimedia/grantmetrics/blob/master/i18n_helper.js.dist (which maybe we could put in our Bundle?). I'm happy to do the work, if you think it's a good idea. Just need the OK on the above copy changes.

samwilson commented:
Okay, all those bits are done. The only thing remaining is to sort out the tutorial messages: are we okay to have two-part messages there? Or are there languages that need to translate those phrases differently and have different bits in larger font?

I've switched the footer links to all be linked on the full phrase, because after the changes above the only one left as a partial link was the licence link, and it looked better to just treat them all the same.

mooeypoo commented:
Sorry, I answered inline at https://github.com/wikimedia/svgtranslate/pull/15#discussion_r228396458

the TL;DR is that I think they should be separate messages, one "heading" and one "description" and we should then figure out the purpose (for qqq) properly for all languages.

samwilson commented:
Cool, so now there's zero HTML in the messages.
